Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,166 @@
+     - parameter bombs: The amount of bombs to randomly place around in the Minefield.
+     **/
+    init(gridSize size: Int, withBombAmount bombs: Int) {
+        
+        let randomPlacements = MineField.getPlacementOptions(count: bombs, forArraySize: size)
+        var cells = [[Cell]]()
+        
+        for x in 0..<size {
+            var yCells = [Cell]()
+            
+            for y in 0..<size {
+                yCells.append(Cell(containsBomb: randomPlacements.contains(where: { placement -> Bool in
+                    return placement == (x: x, y: y)
+                })))
+            }
+            cells.append(yCells)
+        }
+        
+        MineField.markPlacedBombs(inField: &cells, placedBombs: randomPlacements)
+        
+        self.cells = cells
+    }
+    
+    
+    /**
+     Retrieve a cell's immediate neighbours (up, down, left, right & diagonally).
+     - parameter cellCoords: The X & Y position of the cell to retrieve neighbours from.
+     - parameter field: The Minefield to look through.
+     - parameter returns: All the coordinates of the given cell's neighbours (up to 8).
+     **/
+    private static func getImmediateNeighbours(ofCellAt cellCoords: Coordinates, inField field: [[Cell]]) -> [Coordinates] {
+        var neighbouringCellsCoords = [Coordinates]()
+        
+        for xDelta in -1...1 {
+            if let _ = field[safe: xDelta + cellCoords.x] {
+                let x = xDelta + cellCoords.x
+                
+                for yDelta in -1...1 {
+                    if let cell = field[x][safe: yDelta + cellCoords.y], !cell.containsBomb {
+                        neighbouringCellsCoords.append((x: x, y: yDelta + cellCoords.y))
+                    }
+                }
+            }
+        }
+        
+        return neighbouringCellsCoords
+    }
+    
+    
+    /**
+     For any given bomb, increment all neighbour's adjacent bombs count. Only increments cells that do not already contain a bomb.
+     - parameter field: The Minefield's cells to mark.
+     - parameter placedBombs: The coordinates of all the bombs to mark.
+     **/
+    private static func markPlacedBombs(inField field: inout [[Cell]], placedBombs: [Coordinates]) {
+        for placedBomb in placedBombs {
+            let immediateNeighboursCoords = MineField.getImmediateNeighbours(ofCellAt: placedBomb, inField: field)
+            for neighbourCoords in immediateNeighboursCoords {
+                
+                // we can force unwrap here, cells without a bomb
+                field[neighbourCoords.x][neighbourCoords.y].neighbouringBombs! += 1
+            }
+        }
+    }
+    
+    
+    /**
+     Retrieves all coordinates pairs for a given Minefield size.
+     - parameter size: The size of the Minefield.
+     - returns: An Array containing all the coordinates in that Minefield.
+     **/
+    private static func getPlacementOptions(forArraySize size: Int) -> [Coordinates] {
```

Comment:
This function could be replaced with `cells.flatMap { $0 }`

Reply:
Scratch that it would return `[Cell]` rather than co-ordinates, I'm guessing you're returning co-ordinates here to avoid having a copy made.
I think I prefer mutating our `cells` property to returning co-ordinates though.